test/resume/TcpResumeServer.cpp

```diff
@@ -175,35 +142,23 @@ class ServerRequestHandler : public DefaultRequestHandler {
   std::shared_ptr<StreamState> streamState_;
 };
 
-class Callback : public AsyncServerSocket::AcceptCallback {
+class MyServerConnectionAcceptor : public ServerConnectionAcceptor {
  public:
-  Callback(EventBase& eventBase, Stats& stats)
-      : streamState_(std::make_shared<StreamState>(Stats::noop())),
-        eventBase_(eventBase),
-        stats_(stats){};
+  MyServerConnectionAcceptor(EventBase& eventBase, Stats& stats)
+      : ServerConnectionAcceptor(stats), eventBase_(eventBase) {}
 
-  virtual ~Callback() = default;
+  void setupNewSocket(
+      std::shared_ptr<FrameTransport> frameTransport,
+      ConnectionSetupPayload setupPayload) override {
+    CHECK(setupPayload.resumable);
```

Comment:
it seems useful that this server is written as an example that can accept requests from either the resume client or standard client and just fallback to no resumption.  e.g. CHECK seems extreme.